src/common/fam_context.h

```diff
@@ -38,6 +38,10 @@
 #include <rdma/fi_endpoint.h>
 
 #include "common/fam_options.h"
+enum Fam_Req_State {
+	            FAM_REQ_INPROGRESS = 0,
+                    FAM_REQ_COMPLETED = 1,
```

Comment:
You need to add one more state to handle failure requests being processed by a different thread.
FAM_REQ_COMPLETED_WITH_ERR